File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
       
       int minprices=prices[0];
       int maxProfit=0;

       for(int i=1;i<prices.Length;i++){
        if(prices[i]<minprices){
            minprices=prices[i];
        }
           int profit=prices[i]-minprices;
           if(profit>maxProfit){
            maxProfit=profit;
           }
       }

          

        
         return maxProfit;
       }
    
    
    
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
     let minprofit=prices[0];
     let maxProfit=0;
     
     for(i=1;i<prices.length;i++){
         if(prices[i]<minprofit){
            minprofit=prices[i];
         }

         let profit=prices[i]-minprofit;
         if(profit>maxProfit){
            maxProfit=profit;
         }
     }

     return maxProfit;
};